### PR TITLE
Add message author id to ReactionAddEvent 

### PR DESCRIPTION
**Pull request by kezc**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](https://github.com/Javacord/Javacord/blob/master/.github/CONTRIBUTING.md).

## Changelog
- Added message author id to message reaction event

## Description

Closes: https://github.com/Javacord/Javacord/issues/1249

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java [hunk 1]
```diff
@@ -14,4 +14,11 @@ public interface ReactionAddEvent extends SingleReactionEvent {
      */
     CompletableFuture<Void> removeReaction();
 
+    /**
+     * Gets the message author id.
+     *
+     * @return The reaction.
```
Comment (RealYusufIsmail) [inline javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java:20]:
```suggestion
     * @return The message author id.
```

File: javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java [hunk 2]
```diff
@@ -23,15 +28,25 @@ public class ReactionAddEventImpl extends SingleReactionEventImpl implements Rea
...
     @Override
     public CompletableFuture<Void> removeReaction() {
         return Reaction.removeUser(getApi(), getChannel().getId(), getMessageId(), getEmoji(), getUserId());
     }
 
+    /**
+     * Returns the id of the user that sent the message.
+     *
+     * @return The id of the user that sent the message.
+     */
```
Comment (RealYusufIsmail) [inline javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:48]: Java doc is jot required in impls

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

File: javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java (hunk 2 above)
Comment (Vampire) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:48]: That's non-sense.
JavaDoc always makes sense everywhere.
It makes most sense or should always be there on public methods in the public API.
But JavaDoc is always and everywhere better to be there than not to be there.
Thus you should really not motivate people to leave it out, especially when they already added it.

Comment (RealYusufIsmail) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:48]: Ok I apologia

Comment (kezc) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:48]: Anyway, thanks @RealYusufIsmail  for pointing it out as I was forced to add it by linter and I've started wondering why. Now I see that `@Override` annotation is missing.

Comment (Vampire) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:48]: Ah, sorry @RealYusufIsmail, I misread your comment as "you don't need to add JavaDoc to core classes", which are our "implementation" of the logic.

That you don't need it on methods that implement an interface if you properly inherit the JavaDoc is true of course.

Comment (kezc) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java]: Ok, I've removed the unnecessary JavaDoc.

Comment (RealYusufIsmail) [reply javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java]: > classes

Not a problem.

**Review by RealYusufIsmail — APPROVED** [review]
LGTM. Please squash you commits and one of the contrubters will merge your pr if they approve.

File: javacord-core/src/main/java/org/javacord/core/util/handler/message/reaction/MessageReactionAddHandler.java [hunk 3]
```diff
@@ -45,6 +45,7 @@ public void handle(JsonNode packet) {
         long channelId = packet.get("channel_id").asLong();
         long messageId = packet.get("message_id").asLong();
         long userId = packet.get("user_id").asLong();
+        long messageAuthorId = packet.get("message_author_id").asLong();
```
Comment (felldo) [inline javacord-core/src/main/java/org/javacord/core/util/handler/message/reaction/MessageReactionAddHandler.java:48]:
```suggestion
        Long messageAuthorId = packet.hasNonNull("message_author_id") ? packet.get("message_author_id").asLong() : null;
```

File: javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java [hunk 4]
```diff
@@ -14,6 +14,11 @@
  */
 public class ReactionAddEventImpl extends SingleReactionEventImpl implements ReactionAddEvent {
 
+    /**
+     * The id of the user that sent the message.
+     */
+    private final long messageAuthorId;
```
Comment (felldo) [inline javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:20]:
```suggestion
    private final Long messageAuthorId;
```

File: javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java [hunk 5]
```diff
@@ -23,15 +28,21 @@ public class ReactionAddEventImpl extends SingleReactionEventImpl implements Rea
      * @param emoji The emoji.
      * @param userId The id of the user who added the reaction.
      * @param member The member if it happened in a server.
+     * @param messageAuthorId The id of the user that sent the message
      */
-    public ReactionAddEventImpl(
-            DiscordApi api, long messageId, TextChannel channel, Emoji emoji, long userId, Member member) {
+    public ReactionAddEventImpl(DiscordApi api, long messageId, TextChannel channel, Emoji emoji,
+                                long userId, Member member, long messageAuthorId) {
```
Comment (felldo) [inline javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:34]:
```suggestion
                                long userId, Member member, Long messageAuthorId) {
```

File: javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java [hunk 6]
```diff
@@ -23,15 +28,21 @@ public class ReactionAddEventImpl extends SingleReactionEventImpl implements Rea
...
     }
 
     @Override
     public CompletableFuture<Void> removeReaction() {
         return Reaction.removeUser(getApi(), getChannel().getId(), getMessageId(), getEmoji(), getUserId());
     }
 
+    @Override
+    public long getMessageAuthorId() {
+        return messageAuthorId;
```
Comment (felldo) [inline javacord-core/src/main/java/org/javacord/core/event/message/reaction/ReactionAddEventImpl.java:46]:
```suggestion
    public Optional<Long> getMessageAuthorId() {
        return Optional.ofNullable(messageAuthorId);
```

File: javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java [hunk 7]
```diff
@@ -14,4 +14,11 @@ public interface ReactionAddEvent extends SingleReactionEvent {
      */
     CompletableFuture<Void> removeReaction();
 
+    /**
+     * Gets the message author id.
+     *
+     * @return The message author id.
+     */
+    long getMessageAuthorId();
```
Comment (felldo) [inline javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java:22]:
```suggestion
    Optional<Long> getMessageAuthorId();
```
This is an optional field and should return an `Optional<Long>`

**Review by felldo — CHANGES_REQUESTED** [review]

File: javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java (hunk 7 above)
Comment (kezc) [reply javacord-api/src/main/java/org/javacord/api/event/message/reaction/ReactionAddEvent.java]: Good point, thank you!

**Review by felldo — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**